### PR TITLE
Create optional --output parameter for spk infra generate

### LINKED ISSUE #830: `spk infra generate` should have an optional "--output" argument
**As a:**
Developer 

**I want:**
to write the generated directory from `spk infra generate` to a specific path

**Describe the solution you'd like:**
`spk infra generate --output /path/to/generated/output`

**Acceptance Criteria:**
Executing `spk infra generate` with the `--output` argument should write to a path outside of the current working directory.

**Does this require updates to documentation?:**
Yes

### PR DESCRIPTION
**Pull request by yradsmikham**
Close https://github.com/microsoft/bedrock/issues/830.

### CODE REVIEW

**Review by edaena — APPROVED** [review]

**Review by samiyaakhtar — APPROVED** [review]

File: src/commands/infra/generate.ts [hunk 1]
```diff
@@ -32,24 +33,13 @@ export enum DefinitionYAMLExistence {
...
-};
-
 export const execute = async (
   opts: ICommandOptions,
   exitFn: (status: number) => Promise<void>
 ) => {
-  const projectPath = fetchValues(opts);
   const parentPath = process.cwd();
-
+  const projectPath = opts.project || "";
```
Comment (dennisseah) [inline src/commands/infra/generate.ts]: i think it should be 
`const projectPath = opts.project || parentPath;`

Comment (dennisseah) [reply src/commands/infra/generate.ts]: at one point, we check if projectPath and parentPath are the same or not. https://github.com/CatalystCode/spk/blob/master/src/commands/infra/generate.ts#L356

Comment (yradsmikham) [reply src/commands/infra/generate.ts]: Fixed!

**Review by dennisseah — APPROVED** [review]
👍

**Pull request merged**